### PR TITLE
[MODULAR] Fixes mops double acting when cleaning up liquids

### PR DESCRIPTION
**Pull request by Tastyfish**
## About The Pull Request

Makes mops not try to do the default cleaning behavior if soaking up liquids. Slightly counterproductive.

Also makes the liquids_interaction stuff for mops and reagent containers fully modular.

## How This Contributes To The Skyrat Roleplay Experience

Less weird interaction.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>

| Beaker still works |
| :---: |
| image |

| Mop works better |
| :---: |
| image |

</details>

## Changelog

:cl:
fix: Mops no longer try to clean the floor and soak up liquids at the same time.
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/liquids/code/reagent_containers.dm [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+/obj/item/reagent_containers/proc/attack_on_liquids_turf(obj/item/reagent_containers/my_beaker, turf/T, mob/living/user, obj/effect/abstract/liquid_turf/liquids)
+	if(user.combat_mode)
+		return FALSE
+	if(!my_beaker.spillable)
+		return FALSE
+	if(!user.Adjacent(T))
+		return FALSE
+	if(liquids.fire_state) //Use an extinguisher first
+		to_chat(user, "<span class='warning'>You can't scoop up anything while it's on fire!</span>")
```
Comment (Zonespace27) [inline modular_skyrat/modules/liquids/code/reagent_containers.dm]: Span define all of these pls

File: modular_skyrat/modules/liquids/code/reagent_containers.dm [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/obj/item/reagent_containers/Initialize(mapload, vol)
+	. = ..()
+
+	AddComponent(/datum/component/liquids_interaction, TYPE_PROC_REF(/obj/item/reagent_containers/cup/beaker, attack_on_liquids_turf))
+
+/obj/item/reagent_containers/proc/attack_on_liquids_turf(obj/item/reagent_containers/my_beaker, turf/T, mob/living/user, obj/effect/abstract/liquid_turf/liquids)
```
Comment (Zonespace27) [inline modular_skyrat/modules/liquids/code/reagent_containers.dm]: Better var name for T

File: modular_skyrat/modules/liquids/code/reagent_containers.dm [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	if(user.combat_mode)
+		return FALSE
+	if(!my_beaker.spillable)
+		return FALSE
+	if(!user.Adjacent(T))
+		return FALSE
+	if(liquids.fire_state) //Use an extinguisher first
+		to_chat(user, "<span class='warning'>You can't scoop up anything while it's on fire!</span>")
+		return TRUE
+	if(liquids.height == 1)
```
Comment (Zonespace27) [inline modular_skyrat/modules/liquids/code/reagent_containers.dm:25]: (Not asking you to change this unless you want to) good god the inability to fully clean spills sucks

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/liquids/code/reagent_containers.dm (hunk 3 above)
Comment (Tastyfish) [reply modular_skyrat/modules/liquids/code/reagent_containers.dm:25]: The mechanic as designed is that deep spills are scooped with reagent containers and then puddles with a mop.

File: modular_skyrat/modules/liquids/code/reagent_containers.dm (hunk 1 above)
Comment (Tastyfish) [reply modular_skyrat/modules/liquids/code/reagent_containers.dm]: Fixed.

File: modular_skyrat/modules/liquids/code/reagent_containers.dm (hunk 2 above)
Comment (Tastyfish) [reply modular_skyrat/modules/liquids/code/reagent_containers.dm]: Fixed.

**Review by Zonespace27 — APPROVED** [review]

**Pull request merged**